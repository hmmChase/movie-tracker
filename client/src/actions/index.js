export const addMovies = movies => ({
  type: 'ADD_MOVIES',
  movies
});

export const toggleLogin = () => ({
  type: 'TOGGLE_LOGIN'
});

export const storeUserData = ({ id, name }) => ({
  type: 'STORE_USER_DATA',
  id,
  name
});

export const loadFavorites = favorites => ({
  type: 'LOAD_FAVORITES',
  favorites
});