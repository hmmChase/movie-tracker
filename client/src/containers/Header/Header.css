header {
  height: 9rem;
  margin-bottom: 1rem; }

.userControls {
  position: relative;
  bottom: 80%;
  left: 88%;
  display: flex;
  flex-direction: column;
  text-align: right; }
  .userControls .navLink {
    text-decoration: none; }
  .userControls .loginButton,
  .userControls .favoritesButton {
    margin-bottom: 10px; }
  .userControls .favoritesButton {
    background-image: url("https://fontmeme.com/permalink/180521/3eca9b896f474ebbadf5a6e319ec2756.png");
    background-size: 90%; }
  .userControls .favoritesButton:hover {
    background-image: url("https://fontmeme.com/permalink/180521/ccdff501cc41a10b3c92255dced867cb.png");
    background-size: 80%; }
  .userControls .logOutButton {
    background-image: url("https://fontmeme.com/permalink/180521/bd5a24ecf1b03206300104fe5f4018b9.png");
    background-position-y: 80%; }
  .userControls .logOutButton:hover {
    background-image: url("https://fontmeme.com/permalink/180521/8caf3896f2126801e272c59b6e1f8dee.png"); }
  .userControls button {
    display: flex;
    margin-right: 1.5rem; }
  .userControls p {
    position: relative;
    right: 90%;
    margin: 0.5rem 0;
    color: lightblue;
    font-family: 'Permanent Marker';
    font-size: 1.5rem;
    letter-spacing: 1px;
    text-shadow: 2px 2px #2362ce; }

@media only screen and (max-width: 1200px) {
  .userControls button {
    width: 125px;
    background-size: 80%; }
  .userControls p {
    font-size: 1.3rem; }
  .userControls .loginButton {
    background-size: 60%; }
  .userControls .loginButton:hover {
    background-size: 55%; }
  .userControls .logOutButton:hover,
  .userControls .signUpButton:hover {
    background-size: 70%; }
  .userControls .logOutButton {
    background-position-y: 68%; } }

@media only screen and (max-width: 1080px) {
  .userControls {
    left: 86%; } }

@media only screen and (max-width: 960px) {
  header {
    height: 8rem; }
  .headerLogo {
    width: 60%; }
  .userControls p {
    font-size: 1.1rem; }
  .userControls button {
    width: 100px; } }

@media only screen and (max-width: 800px) {
  .userControls {
    flex-direction: row;
    top: 0;
    left: 10%; }
    .userControls p {
      left: 10%; } }

@media only screen and (max-width: 320px) {
  header {
    width: 100%; }
  .userControls {
    bottom: 0;
    left: 0;
    justify-content: center; }
    .userControls .loginButton {
      margin-right: 10px; } }
