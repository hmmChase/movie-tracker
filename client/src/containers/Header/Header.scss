header {
  height: 9rem;
  margin-bottom: 1rem;
}

.userControls {
  position: relative;
  bottom: 80%;
  left: 88%;
  display: flex;
  flex-direction: column;
  text-align: right;

  .navLink {
    text-decoration: none;
  }

  .loginButton,
  .favoritesButton {
    margin-bottom: 10px;
  }

  .favoritesButton {
    background-image: url('https://fontmeme.com/permalink/180521/3eca9b896f474ebbadf5a6e319ec2756.png');
    background-size: 90%;
  }

  .favoritesButton:hover {
    background-image:url('https://fontmeme.com/permalink/180521/ccdff501cc41a10b3c92255dced867cb.png');
    background-size: 80%;
  }

  .logOutButton {
    background-image: url('https://fontmeme.com/permalink/180521/bd5a24ecf1b03206300104fe5f4018b9.png');
    background-position-y: 80%;
  }

  .logOutButton:hover {
    background-image: url('https://fontmeme.com/permalink/180521/8caf3896f2126801e272c59b6e1f8dee.png');
  }

  button {
    display: flex;
    margin-right: 1.5rem;
  }

  p {
    position: relative;
    right: 90%;
    margin: 0.5rem 0;
    color: lightblue;
    font-family: 'Permanent Marker';
    font-size: 1.5rem;
    letter-spacing: 1px;
    text-shadow: 2px 2px #2362ce;
  }
}

@media only screen and (max-width: 1200px) {
  .userControls {
    button {
      width: 125px;
      background-size: 80%;
    }

    p {
      font-size: 1.3rem;
    }

    .loginButton {
      background-size: 60%;
    }

    .loginButton:hover {
      background-size: 55%;
    }

    .logOutButton:hover,
    .signUpButton:hover {
      background-size: 70%;
    }

    .logOutButton {
      background-position-y: 68%;
    }
  }
}

@media only screen and (max-width: 1080px) {
  .userControls {
      left: 86%;
    }
}

@media only screen and (max-width: 960px) {
  header {
    height: 8rem;
  }
  
  .headerLogo {
    width: 60%;
  }

  .userControls {
    p {
    font-size: 1.1rem;
    }

    button {
      width: 100px;
    }
  }
}

@media only screen and (max-width: 800px) {
  .userControls {
    flex-direction: row;
    top: 0;
    left: 10%;

    p {
      left: 10%;
    }
  }
}

@media only screen and (max-width: 320px) {
  header {
    width: 100%;
  }

  .userControls {
    bottom: 0;
    left: 0;
    justify-content: center;

    .loginButton {
      margin-right: 10px;
    }
  }
}