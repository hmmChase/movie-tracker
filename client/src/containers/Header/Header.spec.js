import React from 'react';
import { shallow } from 'enzyme';
import { Header, mapStateToProps, mapDispatchToProps } from './Header';
import { toggleLogin, loadFavorites } from '../../actions/';
jest.mock('../../utils/fetchFavoriteData');

describe('Header', () => {
  let header;
  let mockProps;
  beforeEach(() => {
    mockProps = {
      loggedIn: false,
      name: 'mock name',
      userId: 1,
      toggleLogin: jest.fn(),
      loadFavorites: jest.fn()
    };
    header = shallow(<Header {...mockProps} />);
  });

  it('matches snapshot with user loggedIn true', () => {
    mockProps.loggedIn = true;
    header = shallow(<Header {...mockProps} />);

    expect(header).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should map state properties to props', () => {
      const mockState = {
        user: {
          loggedIn: false,
          name: 'mock name',
          id: 1
        }
      };
      const expected = {
        loggedIn: false,
        name: 'mock name',
        userId: 1
      };
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch on toggleLogin', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = toggleLogin();
      mappedProps.toggleLogin();

      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });

    it('calls dispatch with the correct params on loadFavorites', () => {
      const mockDispatch = jest.fn();
      const mockFavoritesArray = [];
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = loadFavorites(mockFavoritesArray);
      mappedProps.loadFavorites(mockFavoritesArray);

      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });
  });
});
