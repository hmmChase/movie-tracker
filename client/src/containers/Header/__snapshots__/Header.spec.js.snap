// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot with user loggedIn true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    loadFavorites={[Function]}
    loggedIn={true}
    name="mock name"
    toggleLogin={[Function]}
    userId={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/"
>
          <img
                    alt="movie-tracker logo"
                    className="headerLogo"
                    src="https://goo.gl/qAbw5n"
          />
</NavLink>,
        <div
          className="userControls"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/favorites"
          >
                    <button
                              className="favoritesButton"
                              onClick={[Function]}
                    />
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/"
          >
                    <button
                              className="logOutButton"
                              onClick={[Function]}
                    />
          </NavLink>
          <p>
                    Welcome back, 
                    mock name
          </p>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <img
            alt="movie-tracker logo"
            className="headerLogo"
            src="https://goo.gl/qAbw5n"
/>,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "movie-tracker logo",
            "className": "headerLogo",
            "src": "https://goo.gl/qAbw5n",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/favorites"
>
              <button
                            className="favoritesButton"
                            onClick={[Function]}
              />
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/"
>
              <button
                            className="logOutButton"
                            onClick={[Function]}
              />
</NavLink>,
            <p>
              Welcome back, 
              mock name
</p>,
          ],
          "className": "userControls",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": <button
                className="favoritesButton"
                onClick={[Function]}
/>,
              "to": "/favorites",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "favoritesButton",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": <button
                className="logOutButton"
                onClick={[Function]}
/>,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "logOutButton",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Welcome back, ",
                "mock name",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Welcome back, ",
              "mock name",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/"
>
            <img
                        alt="movie-tracker logo"
                        className="headerLogo"
                        src="https://goo.gl/qAbw5n"
            />
</NavLink>,
          <div
            className="userControls"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/favorites"
            >
                        <button
                                    className="favoritesButton"
                                    onClick={[Function]}
                        />
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/"
            >
                        <button
                                    className="logOutButton"
                                    onClick={[Function]}
                        />
            </NavLink>
            <p>
                        Welcome back, 
                        mock name
            </p>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <img
              alt="movie-tracker logo"
              className="headerLogo"
              src="https://goo.gl/qAbw5n"
/>,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "movie-tracker logo",
              "className": "headerLogo",
              "src": "https://goo.gl/qAbw5n",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/favorites"
>
                <button
                                className="favoritesButton"
                                onClick={[Function]}
                />
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/"
>
                <button
                                className="logOutButton"
                                onClick={[Function]}
                />
</NavLink>,
              <p>
                Welcome back, 
                mock name
</p>,
            ],
            "className": "userControls",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": <button
                  className="favoritesButton"
                  onClick={[Function]}
/>,
                "to": "/favorites",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "favoritesButton",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "button",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": <button
                  className="logOutButton"
                  onClick={[Function]}
/>,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "logOutButton",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "button",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Welcome back, ",
                  "mock name",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Welcome back, ",
                "mock name",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
